site/app/models/gradeable/AutogradingVersion.php

```diff
@@ -72,6 +75,58 @@ public function toArray() {
         return $details;
     }
 
+    public function getNonHiddenPoints() {
+        return $this->non_hidden_non_extra_credit + $this->non_hidden_extra_credit;
+    }
+
+    /**
+     * Gets the percent of the possible visible points the submitter earned
+     * @param bool $clamp True to clamp the output to 1
+     * @return float percentage (0 to 1), or NAN if no visible percent
+     */
+    public function getNonHiddenPercent($clamp = false) {
+        $divisor = $this->graded_gradeable->getGradeable()->getAutogradingConfig()->getTotalNonHiddenNonExtraCredit();
+        $dividend = $this->getNonHiddenNonExtraCredit() + $this->getNonHiddenExtraCredit();
+
+        // Avoid divide-by-zero (== not a typo)
+        if($divisor == 0) {
+            return NAN;
+        }
+        $result = floatval($dividend) / $divisor;
+
+        if ($clamp === true && $result > 1.0) {
+            return 1.0;
+        } else if ($result < 0) {
+            return 0.0;
+        }
```

Comment:
This whole block of logic seems to be duplicated multiple times in here and TaGradedGradeable. May want to factor that into its own method.